test/services/school/index.test.js

```diff
@@ -341,6 +341,54 @@ describe('school service', () => {
 				expect(result.isTeamCreationByStudentsEnabled).to.be.false;
 			}
 		);
+		it('should fail to update officialSchoolNumber with wrong format', async () => {
+			const school = await testObjects.createTestSchool({});
+			const admin = await testObjects.createTestUser({
+				schoolId: school._id,
+				roles: ['administrator'],
+			});
+			const params = await testObjects.generateRequestParamsFromUser(admin);
+
+			try {
+				await app.service('/schools').patch(school._id, { officialSchoolNumber: 'foo' }, params);
+			} catch (err) {
+				expect(err).to.not.equal(undefined);
+				expect(err.message).to.include('School ID is incorrect');
+				expect(err.name).to.be.equal('Error');
+			}
+		});
+		it('should succeed to update officialSchoolNumber with correct format', async () => {
+			const school = await testObjects.createTestSchool({});
+			const admin = await testObjects.createTestUser({
+				schoolId: school._id,
+				roles: ['administrator'],
+			});
+			const params = await testObjects.generateRequestParamsFromUser(admin);
+
+			const mockOfficialSchoolNumber = 'BA-13372';
```

Comment:
misleading var name, since in this case we are not mocking.

Reply:
done